frontend/src/app/_components/creator-page/creator-page.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Creator, PreferredLink } from '@app/_models';
+
+@Component({
+  selector: 'app-creator-page',
+  templateUrl: './creator-page.component.html',
+  styleUrls: ['./creator-page.component.scss']
+})
+export class CreatorPageComponent implements OnInit {
+  @Input() creator!: Creator;
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  nameFromLinkType(linkType: string) {
+    switch(linkType) {
+      case 'etsy':
+        return 'Etsy Shop';
+      case 'youtube':
+        return 'YouTube';
+      case 'facebook':
+        return 'Facebook';
+      case 'insta':
+        return 'Instagram';
+      default:
+        return 'Website';
+    }
+  }
+
+  // There's gotta be a better way to do this
+  // this is a messy messy sad typecast nightmare
+  linksToDisplay() {
+    if(this.creator && this.creator.links) {
+      const result : {[_: string]: string | undefined} = Object();
+      for (let k of Object.keys(this.creator.links) as unknown as PreferredLink) {
```

Comment:
I think the right way to do this is just to iterate over all of them and build result.   Like, addToResult(this.creator.links.etsy, 'Etsy Shop'); Just 5 lines like that.

And then if you want to know what their preferred one was to sort it to the top, have a switch (this.creator.links.preferred) with 5 sections.

It's a bit wordy, but it's a lot easier to read than this and probably easier to write than this.

Reply:
noted! v2. that does seem saner.